### PR TITLE
missions interop

### PR DESCRIPTION
**Pull request by alastairdglennie**
don't be scared, it's mostly renaming challenges to missions

### CODE REVIEW

**Review by munroTom — COMMENTED** [review]
Took a look through and things are looking really good! One issue atm is I can't join a public mission using the local firebase setup, I'm guessing because of these firestore rules as my account isn't a moderator?
Not sure if there's a nice way round it?

Screenshot 2020-12-13 at 16 30 56

I also pushed a couple of small commits cause I spotted one or two small things, should have some time this week to take a look at things as well!

File: src/components/ModeratorPage.tsx [hunk 1]
```diff
@@ -65,10 +67,14 @@ const ModeratorPage = ({ user, label, handleClose }: Props) => {
     );
   }
 
-  const reject = (photo: Photo) =>
-    dbFirebase.writeModeration(photo.id, user.id, false);
-  const approve = (photo: Photo) =>
-    dbFirebase.writeModeration(photo.id, user.id, true);
+  const reject = async (photo: Photo) => {
+    await dbFirebase.writeModeration(photo.id, user.id, false);
+    await missions?.refresh();
```
[the remark this answers is not in the document]
Comment (alastairdglennie) [reply src/components/ModeratorPage.tsx:72]: Refresh call triggers the provider to get the update from the databse so the display mission dat updates. E.g. if you press reject, the refresh() means the pending users page will no longer contain that user.

File: src/types/Missions.ts [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+export function isDuplicatingExistingMissionName(
+  missionConfigurableData: ConfigurableMissionData | undefined,
+  existingMissions: MissionFirestoreData[],
+  currentMissionId?: string
+) {
+  if (missionConfigurableData === undefined) {
+    return false;
+  }
+
+  const existingMissionHasName = existingMissions.some(
```
[the remark this answers is not in the document]
Comment (alastairdglennie) [reply src/types/Missions.ts:97]: Just a nice-to-have, happy to remove

**Review by munroTom — APPROVED** [review]
lgtm

**Pull request merged**